pkg/workloads/cortex/serve/serve.py

```diff
@@ -154,17 +154,25 @@ def predict(request: Any = Body(..., media_type="application/json"), debug=False
     debug_obj("payload", request, debug)
     prediction = predictor_impl.predict(request)
 
-    try:
-        json_string = json.dumps(prediction)
-    except Exception as e:
-        raise UserRuntimeException(
-            f"the return value of predict() or one of its nested values is not JSON serializable",
-            str(e),
-        ) from e
-
-    debug_obj("prediction", json_string, debug)
-
-    response = Response(content=json_string, media_type="application/json")
+    if isinstance(prediction, bytes):
+        debug_obj("prediction", prediction, debug)
+        response = Response(content=prediction, media_type="application/octet-stream")
+    elif isinstance(prediction, str):
+        debug_obj("prediction", prediction, debug)
+        response = Response(content=prediction, media_type="text/plain")
+    elif isinstance(prediction, Response):
+        debug_obj("prediction", prediction, debug)
+        response = prediction
+    else:
+        try:
+            json_string = json.dumps(prediction)
+        except Exception as e:
+            raise UserRuntimeException(
+                "the return value of predict() or one of its nested values is not JSON serializable; please return a JSON serializeable object, a bytes object, a string object, or a starlette.response.Response object",
```

Comment:
@deliahu I'm thinking that *"the return value of predict() or one of its nested values is not JSON serializable"* is not necessary as it already appears in `str(e)`. Here's a snapshot of what I get without that:
```
cortex.lib.exceptions.UserRuntimeException: error: please return a JSON serializeable object, a bytes object, a string object, or a starlette.response.Response object: Object of type 'DataFrame' is not JSON serializable: runtime exception
```
The message that's coming from `str(e)` is `Object of type 'InsertTypeHere' is not JSON serializable: runtime exception`, so keeping the above string leads to duplication. 

What do you think?

Reply:
I can see both sides on that one... @vishalbollu wrote that error message originally, so I will let him chime in regarding which he prefers

Reply:
Hmm, so the only change would be the addition of `including its nested fields` text. I see what you mean about the nested fields. 

I have maybe another idea to reduce the verbosity: we know that whatever exception is thrown, you can be sure it comes from the serialization process, so we can just tailor the error message without appending `str(e)`. Thus, the error message could be this: `"<insert type here>" object is not JSON serializable; please return an object that is JSON serializable (including its nested fields), a bytes object, a string, or a starlette.response.Response object`.

What do you think?

Reply:
How would you get the type of the nested value that failed to be serialized? The error message you prepended is fairly close to the output of `str(e)` as you mentioned `Object of type 'DataFrame' is not JSON serializable`

Reply:
You wouldn't. I was thinking that the passed object could happen to be something else other than a dict/list - and that alone would be relevant. But yeah, you lose that key information of which nested field failed in the case of a dict/list. 

Maybe place `str(e)` before the custom message? Here's a sample: `cortex.lib.exceptions.UserRuntimeException: error: Object of type 'DataFrame' is not JSON serializable; please return an object that is JSON serializable (including its nested fields), a bytes object, a string, or a starlette.response.Response object: runtime exception`

Reply:
Sure ship it!

Reply:
Got it! Tested the changes and it's good to go.